src/main/java/com/minecolonies/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -111,9 +111,10 @@ protected AbstractEntityAIStructure(@NotNull final J job)
      */
     private Supplier<AIState> generateSchematicIterator(Function<Structure.SchematicBlock, Boolean> evaluationFunction, AIState nextState)
     {
-        // do not replace with method reference, this one stays the same on changing reference for currentStructure
-        Supplier<Structure.SchematicBlock> getCurrentBlock = currentStructure::getCurrentBlock;
-        Supplier<Structure.Result>         advanceBlock    = currentStructure::advanceBlock;
+        //do not replace with method reference, this one stays the same on changing reference for currentStructure
+        //URGENT: DO NOT REPLACE FOR ANY MEANS THIS WILL CRASH THE GAME.
+        Supplier<Structure.SchematicBlock> getCurrentBlock = () -> currentStructure.getCurrentBlock();
+        Supplier<Structure.Result>         advanceBlock    = () -> currentStructure.advanceBlock();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1612)